modules/aws-bastion-instance/main.tf

```diff
@@ -101,43 +96,6 @@ resource "aws_launch_configuration" "bastion" {
   }
 }
 
-resource "aws_autoscaling_group" "bastion" {
-  name = aws_launch_configuration.bastion.name
-
-  vpc_zone_identifier = data.aws_subnet_ids.vpc.ids
-
-  desired_capacity          = 1
-  min_size                  = 1
-  max_size                  = 1
-  health_check_grace_period = 60
-  health_check_type         = "EC2"
-  force_delete              = false
-  wait_for_capacity_timeout = 0
-  launch_configuration      = aws_launch_configuration.bastion.name
-
-  enabled_metrics = [
-    "GroupMinSize",
-    "GroupMaxSize",
-    "GroupDesiredCapacity",
-    "GroupInServiceInstances",
-    "GroupPendingInstances",
-    "GroupStandbyInstances",
-    "GroupTerminatingInstances",
-    "GroupTotalInstances",
-  ]
-
-  tags = concat(
-  [
-    {
-      key                 = "Name",
-      value               = var.name,
-      propagate_at_launch = true
-    }
-  ],
-  var.extra_tags
-  )
-
-  lifecycle {
-    create_before_destroy = true
-  }
+resource "aws_eip" "bastion" {
```

Comment:
Improvement in case you already have Elastic IP:

```tf
resource "aws_eip" "bastion" {
  count = var.eip_id ? 0 : 1

  instance = aws_instance.bastion.id
}

resource "aws_eip_association" "bastion" {
  count = var.eip_id ? 1 : 0

  instance_id   = aws_instance.bastion.id
  allocation_id = var.eip_id
}
```

Modified outputs:
```
variable "bastion_ip" {
  value = var.eip_id ? aws_eip_association.bastion[0].public_ip : aws_eip.bastion[0].public_ip
}
```

Reply:
Good point.